app.py

```diff
@@ -0,0 +1,423 @@
+                self.log_debug("Wrote thumbnail %s" % thumbnail_jpg)
+                self.sgtk.shotgun.upload_thumbnail(sg_data["type"], sg_data["id"], thumbnail_jpg)
+                self.log_debug("Uploaded thumbnail to shotgun.")
+                try:
+                    os.remove(thumbnail_jpg)
+                    self.log_debug("Removed temporary file '%s'." % thumbnail_jpg)
+                except Exception, e:
+                    self.log_warning("Could not remove temporary file '%s': %s" % (thumbnail_jpg, e))    
+                
+            # thumbnail upload done!
+
+        # now start the version creation process
+        self.log_debug("Will associate upload with shotgun entity %s..." % sg_data)
+
+        # create a version in Shotgun
+        if info["versionNumber"] != 0:
+            title = "%s v%03d" % (info["sequenceName"], info["versionNumber"])
+        else:
+            title = info["sequenceName"]
+        
+        data = {}
+        data["code"] = title
+        data["description"] = comments
+        data["project"] = self.context.project
+        data["entity"] = sg_data
+        data["created_by"] = self.context.user
+        data["user"] = self.context.user
+        
+        # general metadata for the version
+        # for the frame range, there isn't very meaningful metadata we can add
+        # and we don't have corresponding frames on disk
+        # so set the first frame to 1 in order to normalize the frames from flame
+        # which typically start at 10:00:00.00
+        #
+        # also note that flame is out-exclusive, meaning that if you have the 
+        # frame range 100-111, it corresponds to the frames
+        # 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110
+        # 
+        # We transform the above frame range (100-111) to be 1-10 in shotgun with length 10. 
+        #
+        data["sg_first_frame"] = 1
+        data["sg_last_frame"] = info["sourceOut"] - info["sourceIn"]
+        data["frame_count"] = info["sourceOut"] - info["sourceIn"]
+        data["frame_range"] = "%s-%s" % (data["sg_first_frame"], data["sg_last_frame"])         
+        data["sg_frames_have_slate"] = False
+        data["sg_movie_has_slate"] = False 
+        data["sg_frames_aspect_ratio"] = info["aspectRatio"]
+        data["sg_movie_aspect_ratio"] = info["aspectRatio"]
+        
+        # This is used to find the latest Version from the same department.
+        # todo: make this configurable?
+        data["sg_department"] = "Editorial" 
+        
+        sg_version_data = self.shotgun.create("Version", data)
+        
+        self.log_debug("Created a version in Shotgun: %s" % sg_version_data)
+        
+        # upload quicktime to Shotgun
+        self.log_debug("Begin upload of quicktime to shotgun...")
+        self.shotgun.upload("Version", sg_version_data["id"], full_path, "sg_uploaded_movie")
+        self.log_debug("Upload complete!")
+        
+        # clean up
+        try:
```

Comment:
May be this is the intended behaviour, but quicktimes will be left behind if there is any problem before this point. May be this should be in a finally clause, ensuring clean up is always done ?